vg/recorder/recorder.go

```diff
@@ -0,0 +1,458 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package recorder provides support for vector graphics serialization
+// it is intended to be used as a testing helper.
+package recorder
+
+import (
+	"fmt"
+	"image/color"
+	"runtime"
+
+	"github.com/gonum/plot/vg"
+)
+
+var _ vg.Canvas = (*Canvas)(nil)
+
+// Canvas implements vg.Canvas operation serialization.
+type Canvas struct {
+	// Resolution holds the canvas resolution in DPI.
+	Resolution float64
+
+	// Actions holds the complete state of the canvas.
+	Actions []Action
+
+	// KeepCaller indicates whether the Canvas will
+	// retain caller information for the actions.
+	KeepCaller bool
+}
+
+// Action is a vector graphics action as defined by the
+// vg.Canvas interface. Each method of vg.Canvas must have
+// a corresponding Action type.
+type Action interface {
+	VgCall() string
+	ApplyTo(vg.Canvas)
+	callRecorder() *callRecorder
+}
+
+type callRecorder struct {
```

Comment:
I'll try and find something better than callRecorder; record is not informative IMO.